core/src/main/java/com/oath/oak/Chunk.java

```diff
@@ -501,21 +589,26 @@ int linkEntry(int ei, K key) {
     /**
      * write value in place
      **/
-    void writeValue(int hi, V value) {
-        assert hi >= 0;
-        ByteBuffer byteBuffer = memoryManager.allocate(valueSerializer.calculateSize(value));
-        // just allocated bytebuffer is ensured to have position 0
-        valueSerializer.serialize(value, byteBuffer.slice());
-        handles[hi].setValue(byteBuffer);
+    long writeValue(V value) {
+        // the length of the given value plus its header
+        int valueLength = valueSerializer.calculateSize(value) + ValueUtils.VALUE_HEADER_SIZE;
+        Slice slice = memoryManager.allocateSlice(valueLength);
+        // initializing the header lock to be free
+        slice.initHeader();
+        // since this is a private environment we can only use slice, instead of duplicate and then slice
```

Comment:
There is an ambiguity between our class Slice and ByteBuffer method slice()... Can you please make it clearer?